raft/raft_log.hpp

```diff
@@ -45,17 +49,27 @@ namespace bzn
 
         size_t size() const;
 
+        inline size_t memory_used() const {return this->total_memory_used;};
+
         inline const std::vector<log_entry>& get_log_entries()
         {
             return this->log_entries;
         }
 
 
+        inline bool maximum_storage_exceeded() const
+        {
+            return this->maximum_storage < this->total_memory_used;
+        }
+
     private:
         void append_log_disk(const bzn::log_entry& log_entry);
         void rewrite_log();
 
         std::vector<log_entry> log_entries;
+        size_t                  total_memory_used = 0;
+        size_t                  maximum_storage;
```

Comment:
this should be a const and initialized at construction?

Reply:
Yes. Changes maximum_storage to a const. It was already being initialized at construction.